### ISSUE TITLE
Bug in computing hypergeometric traces

### ISSUE THREAD

**Issue by kedlaya** [sagemath/sage#37910]
### Steps To Reproduce

```
sage: from sage.modular.hypergeometric_motive import HypergeometricData as HGData
sage: H = HGData(alpha_beta=[[1/2,1/3,2/3,1/6,5/6], [0,0,0,0,0]])
sage: H.padic_H_value(6211, 2, -512000)
```

### Expected Behavior

Should return `-184336511256141`.

### Actual Behavior

```
----> 9 cpdef hgm_coeffs(long long p, int f, int prec, gamma, m, int D,
     10                  gtable, int gtable_prec, bint use_longs):
     11     r"""
File /usr/local/sage/src/sage/modular/hypergeometric_misc.pyx:73, in sage.modular.hypergeometric_misc.hgm_coeffs()
     71     except TypeError:
     72         for r in range(q1):
---> 73             gtab2[r] = gtable[r].lift()
     74 if f == 1:
     75     for r in range(q1):
OverflowError: int too big to convert
```

### Environment

```markdown
- **OS**: Ubuntu 24.04
- **Sage Version**: 10.4b4
```

### Checklist

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.
- [X] I have read the documentation and troubleshoot guide

**Comment by kedlaya** [comment]
The issue here is that `hgm_coeffs()` is assuming `p ** prec < 2 ** 63` in order to fit each Gauss sum into a `long long`.

**Comment by tscrim** [comment]
Fixed by #37911.
